app/main.py

```diff
@@ -2,0 +1,42 @@
+from __future__ import annotations
+
+
+class Animal:
+
+    alive = []
+
+    def __init__(
+            self,
+            name: str,
+            health: int = 100,
+            hidden: bool = False
+    ) -> None:
+        self.name = name
+        self.health = health
+        self.hidden = hidden
+
+        if self.health > 0:
+            Animal.alive.append(self)
+
+    def __repr__(self) -> str:
+        return str((
+            f"{{Name: {self.name}, "
+            f"Health: {self.health}, "
+            f"Hidden: {self.hidden}}}"
+        ))
+
+
+class Herbivore(Animal):
+
+    def hide(self) -> None:
+        self.hidden = not self.hidden
+
+
+class Carnivore(Animal):
+
+    @staticmethod
+    def bite(animal: Herbivore | Carnivore) -> None:
```

Comment:
The type annotation `Herbivore | Carnivore` is correct for Python 3.10 and later. If you need to support earlier versions, consider using `Union[Herbivore, Carnivore]` from the `typing` module.